### ISSUE TITLE
Repetited text within and between segments

### ISSUE THREAD

**Issue by cxumol** [guillaumekln/faster-whisper#40]
Sentence sometimes repeat itself for multiple times at the end of a segment and/or continue to repeat to their next segments.

This was a known issue of openai/whisper (upstream issue https://github.com/openai/whisper/discussions/977 and https://github.com/openai/whisper/discussions/1059), and may be fixed by https://github.com/openai/whisper/commit/38f2f4d99d297c2fc09f9f5f28eaa06b017e24a3

When I use "faster-whisper", I encountered the same sentence repetition. I found it's also reported on https://github.com/guillaumekln/faster-whisper/issues/35#issuecomment-1464840201

Could you please check if this commit https://github.com/openai/whisper/commit/38f2f4d99d297c2fc09f9f5f28eaa06b017e24a3 can/should be ported here?

**Comment by cxumol** [comment]
yes this one https://github.com/openai/whisper/commit/671ac5a4ceb8980403554da94feb995161fb5fef make more sense within the segment level

**Comment by guillaumekln** [comment]
This commit was already ported in https://github.com/guillaumekln/faster-whisper/commit/6a84df400fb7bc3ff7fc2197a4586e9338d981bb.

Can you check you are using the latest version of the library? If yes, it would be great if you can share the audio file and parameters that trigger the issue.

**Comment by shumeirao** [comment]
> This commit was already ported in [6a84df4](https://github.com/guillaumekln/faster-whisper/commit/6a84df400fb7bc3ff7fc2197a4586e9338d981bb).
> 
> Can you check you are using the latest version of the library? If yes, it would be great if you can share the audio file and parameters that trigger the issue.

I also encountered this situation.ctranslate2 version is 3.8,predict code is:segments, info = model_multi.transcribe(audio_file,beam_size=5,language=language)

**Comment by guillaumekln** [comment]
Your audio triggers the "temperature fallback" so the transcription will change each time you run the model. Since the results are random it can sometimes contain repetitions. Do you think the repetitions occur more often in faster-whisper than in openai/whisper?

**Comment by cxumol** [comment]
I don't think repetition is derived from "temperature fallback". Because when I manually set `temperature=0` I could observe even more repetition sometimes.

**Comment by cxumol** [comment]
I used "jonatasgrosman/whisper-large-zh-cv11" model to transcribe https://justpodmedia.com/audio/left-right/leftright-ep179-20220418.mp3

![image](https://user-images.githubusercontent.com/8279655/225807220-756ccb85-72bd-4985-9ac2-4b898e9d718f.png)

Repeated text is underlined. The length of the marked segment [4556.26s -> 4556.52s] is 0.24s. It's not reasonable to speak so long within 0.24s.
